effectful-core/src/Effectful/Internal/Monad.hs

```diff
@@ -190,14 +198,24 @@ concUnliftEff
   -> Limit
   -> ((forall r. Eff es r -> IO r) -> IO a)
   -> IO a
-concUnliftEff es Ephemeral (Limited uses) k =
-  ephemeralConcUnliftIO uses k es unEff
-concUnliftEff es Ephemeral Unlimited k =
-  ephemeralConcUnliftIO maxBound k es unEff
-concUnliftEff es Persistent (Limited threads) k =
-  persistentConcUnliftIO False threads k es unEff
-concUnliftEff es Persistent Unlimited k =
-  persistentConcUnliftIO True maxBound k es unEff
+concUnliftEff es persistence limit k = concUnliftIO persistence limit k es unEff
+
+-- | Create an unlifting function with the current 'UnliftStrategy'.
+--
+-- This function is equivalent to 'withRunInIO', but has a 'HasCallStack'
+-- constraint for accurate stack traces in case an insufficiently powerful
+-- 'UnliftStrategy' is used and the unlifting function fails.
+--
+-- /Note:/ the strategy is reset to 'SeqUnlift' for new threads.
```

Comment:
Because we reset the unlifting strategy in `withEffToIO` and not in `withUnliftStrategy`. I got hit by this when I first wrote a test for `withUnliftStrategy` alone (not sure why I didn't include it in this PR) which in turned failed because nothing was reset.